### PR TITLE
Fix dynamiccontroller returned error missing

### PR DESCRIPTION
**Pull request by Abirdcfly**
Signed-off-by: Abirdcfly <fp544037857@gmail.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The dynamiccontroller need return raw error to caller.Otherwise caller use kubernetes built-in methods like `errors.IsNotFound(err)` will get unexpected result.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/dynamiccontroller/application/application.go [hunk 1]
```diff
@@ -477,21 +476,21 @@ func (c *Center) Process(msg model.Message) {
...
-			c.Response(app, msg.GetID(), Approved, "", retObj)
+			c.Response(app, msg.GetID(), Approved, nil, retObj)
 		default:
 			return fmt.Errorf("unsupported Application Verb type :%v", app.Verb)
 		}
 		return nil
 	}()
 	if err != nil {
-		c.Response(app, msg.GetID(), Rejected, err.Error(), nil)
+		c.Response(app, msg.GetID(), Rejected, err,nil)
```
Comment (fisherxu) [inline cloud/pkg/dynamiccontroller/application/application.go:486]:
```suggestion
		c.Response(app, msg.GetID(), Rejected, err, nil)
```

**Comment by fisherxu** [comment]
> /hold There has a bug...test ci should not pass...I guess we need more e2e test for dynamiccontroller😂

Yes, now we still haven't the ci for dynamiccontroller/mateserver.

File: cloud/pkg/dynamiccontroller/application/application.go [hunk 2]
```diff
@@ -4,11 +4,10 @@ import (
 	"context"
 	"crypto/md5"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"sync"
 	"time"
-
+
```
Comment (Iceber) [inline cloud/pkg/dynamiccontroller/application/application.go:10]: please format

Comment (Abirdcfly) [reply cloud/pkg/dynamiccontroller/application/application.go:10]: Got it. Thx

File: cloud/pkg/dynamiccontroller/application/application.go (hunk 1 above)
Comment (Abirdcfly) [reply cloud/pkg/dynamiccontroller/application/application.go]: Got it, Thx

**Comment by GsssC** [comment]
Start to review. Changing application.Reason to adapt api errors is in evaluation.
Originally just want to use application.Reason to briefly describe why app is in application status, and the scope is bigger than returned error from APIServer, e.g. application is rejected by Application Center before calling APIServer(An authentication layer that must to do but have not yet).
: )

**Pull request closed without merging**